### PR TITLE
ARTnet: addition of init.hiv.prev parameter

### PR DESCRIPTION
**Pull request by andsv2**
Addition of init.hiv.prev parameter.

### CODE REVIEW

**Comment by andsv2** [comment]
@smjenness can you review these changes for merging into master branch:

* Addition of `init.hiv.prev` input parameter. Default of NULL; vector of size 1 or 3 otherwise.
* Changes to ARTnet intro documentation to reflect above.
* Change to `race = FALSE` as default for `build_epistats`.

File: R/NetStats.R [hunk 1]
```diff
@@ -168,16 +168,40 @@ build_netstats <- function(epistats, netparams,
...
+
+      attr_diag.status[samp.B] <- rbinom(length(samp.B), 1, init.hiv.prev[1])
+      attr_diag.status[samp.H] <- rbinom(length(samp.H), 1, init.hiv.prev[2])
+      attr_diag.status[samp.W] <- rbinom(length(samp.W), 1, init.hiv.prev[3])
+
+      out$attr$diag.status <- attr_diag.status
+
+    } else {
+    init.hiv.prev <- epistats$init.hiv.prev
+    samp.size <- ceiling(network.size*init.hiv.prev)
```
Comment (smjenness) [inline R/NetStats.R]: As previously discussed, you need to use standardized code styling (e.g., this block should be indented)

File: R/EpiStats.R [hunk 2]
```diff
@@ -420,47 +427,67 @@ build_epistats <- function(geog.lvl = NULL, geog.cat = NULL, race = TRUE,
...
+        init.hiv.prev <- init.hiv.prev[1]
+      }
     }
   }
 
   # Save Out File -----------------------------------------------------------
 
-  out <- list()
+  if(is.null(geog.lvl)){
+    out$geogYN.l <- l$geogYN
```
Comment (smjenness) [inline R/EpiStats.R]: non-standard code styling

File: R/EpiStats.R [hunk 3]
```diff
@@ -420,47 +427,67 @@ build_epistats <- function(geog.lvl = NULL, geog.cat = NULL, race = TRUE,
...
-
-      hiv.mod <- glm(hiv2 ~ age ,
-                     data = d1, family = binomial())
-    } else {
-      d1 <- select(d, geogYN, age, hiv2)
-
-      hiv.mod <- glm(hiv2 ~ age + geogYN,
-                     data = d1, family = binomial())
+    if (race == TRUE){
+      if(length(init.hiv.prev) != 3 ){
```
Comment (smjenness) [inline R/EpiStats.R]: non-standard code styling

File: R/EpiStats.R [hunk 4]
```diff
@@ -13,6 +13,11 @@
 #'        (35, 45], (45, 55], (55, 65], (65, 100].
 #' @param race Whether to stratify by racial status. Default is TRUE.
 #' @param browser Run function in interactive browser mode. Default is FALSE.
+#' @param init.hiv.prev Initial HIV prevalence of estimated model, vector of size
+#'.        pertaining to prevalence among three racial classes (black, hispanic and
+#'         white respectively). If "NULL", ARTnet will handle calculation of
+#'         prevalence through ARTnet data. if `race = FALSE`, will take first element
+#'         as of vector as overall HIV prevalence.
 #'
```
Comment (smjenness) [inline R/EpiStats.R]: No, let's allow for specifying three different prevalence values even if race = FALSE.

File: R/EpiStats.R [hunk 5]
```diff
@@ -13,6 +13,11 @@
 #'        (35, 45], (45, 55], (55, 65], (65, 100].
 #' @param race Whether to stratify by racial status. Default is TRUE.
 #' @param browser Run function in interactive browser mode. Default is FALSE.
+#' @param init.hiv.prev Initial HIV prevalence of estimated model, vector of size
+#'.        pertaining to prevalence among three racial classes (black, hispanic and
+#'         white respectively). If "NULL", ARTnet will handle calculation of
+#'         prevalence through ARTnet data. if `race = FALSE`, will take first element
```
Comment (smjenness) [inline R/EpiStats.R]: NULL isn't a character vector. As previously discussed, all code in the documentation should be wrapped in `\code{x]`.

File: vignettes/ARTnetIntro.R [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+## ------------------------------------------------------------------------
+#Set age limits
+epistats8 <- build_epistats(geog.lvl = "state", geog.cat = "GA", 
+                            race = TRUE, age.limits = c(15, 65),
+                            age.breaks = c(20, 30, 40, 50, 60),
+                            init.hiv.prev = c(0.17,0.1,0.13))
+
+#Specify age categories: (0-20], (20, 30], . . . (60, 100]
+netparams1 <- build_netparams(epistats = epistats8, smooth.main.dur = TRUE)
+
```
Comment (smjenness) [inline vignettes/ARTnetIntro.R]: please add these all into unit tests as well that checks for consistency of input and expected values

**Comment by smjenness** [comment]
@andsv2 What is the status of this PR at this point?

**Comment by andsv2** [comment]
@smjenness afternoon Sam: the final change that was/is being debugged at the moment is the suggested change to ASMR to account for scenarios with reduced age ranges. This is easily done, but it breaks the default `departures.FUN` in EpiModelHIV: right now `departures.net` pulls the correct mortality rate based on the age of the individual; with the new change this would need to be changed to account for the new row-assignments based on age limitations. My question then is: is this needed? ASMR as used is not affected by age parameters that are passed through ARTnet. If it is, then EpiModelHIV will need to be changed.

**Comment by smjenness** [comment]
Yes, please update the departure function so it works with the new age range specifications. Ensure that this is performing correctly by debugging the code and demonstrate the validity using metrics.

**Comment by andsv2** [comment]
@smjenness this is ready to be merged. If okay I can do.

**Comment by smjenness** [comment]
Have you tested that this works against all the various combinations of inputs?

**Comment by andsv2** [comment]
> Have you tested that this works against all the various combinations of inputs?

@smjenness: ran it through the ARTnet Intro as well as some other city/region combos and stats are internally consistent.

**Pull request merged**